polyseam/cndi
# [Enhancement]: Provision SSH key up front

# Description

Until now we have relied on each cloud to provision SSH keys for us, and this has served us well, but when attempting to safely remove a node from the cluster when required we would benefit from having access to the key to do so. With this PR users will also be able to seamlessly access their nodes without ever logging into the cloud console.

All that being said, we still do not recommend users log into VMs unless absolutely necessary, and the presence of the keypair should not imply otherwise.

# Test Instructions

- `cndi-next init -i -d -t ec2/neo4j --workflow-ref gen-ssh-keys-upfront`
- git add, commit, push
- wait...
- `ssh -i ssh_access_key.pem my-node-public-host.ec2.blah.aws.com`
- see that we are able to log in successfully

# Code of Conduct

By submitting this Pull Request, you agree to follow our
[Code of Conduct](https://github.com/polyseam/cndi/blob/main/CODE_OF_CONDUCT.md)


# Notes (Optional)
